app/src/core/utils/apolloError.ts

```diff
@@ -50,121 +64,83 @@ export const TryGetApolloErrorEntity = (error: ApolloError): string | null => {
     }
     return null;
 };
-export const getStatusCode = (error: ApolloError): number | null => (error?.networkError as ServerError)?.statusCode;
+
+export const isCodeInHTTPErrorCodes = (code: number): code is StatusCodeType => {
+    return Object.values(StatusCodes).includes(code as StatusCodeType);
+};
+
+export const isCodeA400Error = (code: number): code is StatusCodes400 => {
+    return code > 399 && code < 500;
+};
+
+export const getStatusCode = (error: ApolloError): Result<StatusCodeType, ApolloError> => {
+    if (guard<ServerError>(error.networkError, ['statusCode'])) {
+        const statusCode = error.networkError.statusCode;
+        if (isCodeInHTTPErrorCodes(statusCode)) {
+            return ok(statusCode);
+        } else {
+            return err(error);
+        }
+    } else {
+        return err(error);
+    }
+};
 
 // TODO: Want to come up with a better pattern for error handling than these error specific handlers. At some point the decision
```

Comment:
I think you did this. Want to remove this or leave it in?

Reply:
Oh no. You kinda removed the right logic and made it more generic. Maybe I should reread it